src/test/java/org/example/service/TicketServiceTest.java

```diff
@@ -0,0 +1,223 @@
+package org.example.service;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.example.model.TicketType;
+import org.example.model.UserStatus;
+import org.example.model.ticket.Ticket;
+import org.example.model.user.User;
+import org.example.repository.TicketRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.dao.OptimisticLockingFailureException;
+
+@ExtendWith(MockitoExtension.class)
+class TicketServiceTest {
+
+  @Mock private TicketRepository ticketRepository;
+  @InjectMocks private TicketService ticketService;
+
+  User user;
+  Ticket testTicket;
+
+  @BeforeEach
+  void init() {
+    user = new User("Masha", UserStatus.DEACTIVATED, new Timestamp(System.currentTimeMillis()));
+    testTicket = new Ticket(user, TicketType.DAY, new Timestamp(System.currentTimeMillis()));
+  }
+
+  @Test
+  void saveTicket_withTicket() {
+    testTicket.setId(UUID.randomUUID());
+
+    when(ticketRepository.save(testTicket)).thenReturn(testTicket);
+
+    Ticket result = ticketService.saveTicket(testTicket);
+
+    assertNotNull(result);
+    assertNotNull(testTicket.getId());
```

Comment:
this assertion is redundant, it is better to compare id with the following approoach:
```java
    UUID expectedId = UUID.randomUUID();
    testTicket.setId(expectedId);

    when(ticketRepository.save(testTicket)).thenReturn(testTicket);

    Ticket actualTicket = ticketService.saveTicket(testTicket);

    assertNotNull(actualTicket);
    assertEquals(expectedId, actualTicket.getId());
    verify(ticketRepository, times(1)).save(testTicket);
```

Reply:
Changed code for saveTicket_withTicket() test